### ISSUE TITLE
[decaying] Make `last_seen` takes precedence over timestamp for the decay

### ISSUE THREAD

**Issue by mokaddem** [MISP/MISP#5836]
The way I see it for the moment would be something like
```
if last_sighting > timestamp:
    use last_sighting
else:
    if `last_seen` is null:
        use `timestamp`
    else: # we consider that timestamp >= last_seen
        use (`last_sighting` > `last_seen` ? `last_sighting` : `last_seen`)
```

**Comment by andurin** [comment]
TL;DR:
With fs/ls entries the timestamp gets more a "last changed" time. With that in mind givving a precedence on last_seen over timestamp to calculate the decay is a "GoodIdea" IMO.

Long Version: 
In my opinion decaying follows the goal that a specific dangerous indicator may not be that dangerous in the future. One may abuse some phishing sites down, advisories give their hosts up, etc. So it's interesting to bring this picture into misp - decay feature for the rescue! 
As long as you can determine "the bad" as existing (aka sighting) the attributes keeps its high dangerous state. 
As soon as there are no new sightings the aging out phase begins. Since changes to tags will update the timestamp (last changed character) the aging out would be stopped (Bad Idea if you just change the Phishing URL to "inactive" by Tag change). This doesn't mean that you see that attribute again, only that you've done a re-classification.

**Comment by andurin** [comment]
Open Question(s):
Does a sighting will automatically update last_seen even if it wasn't set before?

**Comment by mokaddem** [comment]
> Open Question(s):
> Does a sighting will automatically update last_seen even if it wasn't set before?

No. It won't but if the `last_sighting > timestamp`, `last_sighting` is used. So for the decay, adding sightings is 'safe' in that sense.

**Comment by mokaddem** [comment]
The fact that adding/removing tags reset the decay of an attribute was an unplanned side effect. The initial design was that the decay should be reset if an attribute changes, because it's not the same attribute anymore. And when I implemented it, I thought that changing tags would change the meaning of the attribute, therefore requiring a new decay.
Your example with setting an `inactive` was covered with the `retention:expired` tag. You *just* have to know that other formula exist and what they do..
In the end, I also think that `last_seen` should take precedence over the `timestamp` is set.

**Comment by iglocska** [comment]
Absolutely, timestamp should always be a fallback if no other timestamp is provided. Timestamp first and foremost is a last edit field - which we can use in lieu of anything better, but it should be a last resort.
